src/containers/exchange/components/CancelOrderButton.js

```diff
@@ -69,22 +74,33 @@ class CancelOrderButton extends React.Component {
                 {label}
             </a>
         );
-        return !order ? null : (
+        return (
```

Comment:
The issue I was fixing that shortly after submission the modal closed then after a few secs the success modal opened.  It was because the ordercancel event arrived sooner sometimes than the tx success event.  On order cancel (or fill) events we releod orders which caused the component to rerender with null order.  
Any better way to do it?
Btw, It wouldn't help if we maintain orders state as with transfers, the order would disappear.. See [trello card](https://trello.com/c/T6hoxgTg/66-maintain-orders-state) I just added

Reply:
I guess adding something generic to the store that gives information about the order status could help.

Reply:
we could. wondering whats the issue we are trying to solve? any issue with current behaviour or method?

Reply:
while you were doing the refactoring there were some race conditions in the data coming from different places. To avoid that it would be the best if we keep everything in the state and use a selector like [reselect](https://github.com/reactjs/reselect) to retrieve. This way all the dependencies are handled in one place.

Reply:
got you. created a [trello card](https://trello.com/c/nA6deQMe/77-refactor-eventshandling-refresh) for discussion.